spectrwm.c

```diff
@@ -9148,9 +9148,9 @@ setconfvalue(const char *selector, const char *value, int flags)
 		if (getenv("SWM_STARTED") != NULL)
 			return (0);
 
-		bzero(s, sizeof s);
-		if (sscanf(value, "ws[%d]:%1023c", &ws_id, s) != 2)
+		if (sscanf(value, "ws[%d]:%n", &ws_id, &n) != 1 || !value[n])
```

Comment:
If `value` is too short to match `%n`, `%d` could still be matched (with sscanf returning 1) and `n` will not be assigned a value. The `value[n]` test would be invalid.

Also, openbsd [style(9)](http://man.openbsd.org/OpenBSD-current/man9/style.9) states:

> Don't use ‘!’ for tests unless it's a boolean, i.e., use
> if (*p == '\0')
> not
> if (!*p)

Reply:
Sorry, that was indeed sloppy. Should be fixed now. :sweat_smile: